packages/react-router-dom/__tests__/data-browser-router-test.tsx

```diff
@@ -2915,7 +2993,7 @@ function testDomRouter(
           let { container } = render(<RouterProvider router={router} />);
 
           expect(container.querySelector("form")?.getAttribute("action")).toBe(
-            "/foo"
+            "/foo/bar"
```

Comment:
This existing test was asserting the buggy behavior.  When `action="."` is specified it should inherit the current path including the splat portion